packages/react/src/dialog/root/DialogRoot.test.tsx

```diff
@@ -44,43 +44,93 @@ describe('<Dialog.Root />', () => {
       setProps({ open: false });
       expect(queryByRole('dialog')).to.equal(null);
     });
-  });
 
-  // toWarnDev doesn't work reliably with async rendering. To re-eanble after it's fixed in the test-utils.
-  // eslint-disable-next-line mocha/no-skipped-tests
-  describe.skip('prop: modal', () => {
-    it('warns when the dialog is modal but no backdrop is present', async () => {
-      await expect(() =>
-        render(
-          <Dialog.Root modal animated={false}>
-            <Dialog.Popup />
-          </Dialog.Root>,
-        ),
-      ).toWarnDev([
-        'Base UI: The Dialog is modal but no backdrop is present. Add the backdrop component to prevent interacting with the rest of the page.',
-        'Base UI: The Dialog is modal but no backdrop is present. Add the backdrop component to prevent interacting with the rest of the page.',
-      ]);
-    });
+    it('should remove the popup when animated=true and there is no exit animation defined', async function test(t = {}) {
```

Comment:
We can create a reusable test suite for veryfing behavior of popup-like components in the future, so we don't need to copy the same tests several times.